lib/core/request.js

```diff
@@ -150,19 +156,27 @@ class Request {
 
     this.headers = ''
 
+    this.expectContinue = expectContinue != null ? expectContinue : false
+
     if (Array.isArray(headers)) {
       if (headers.length % 2 !== 0) {
         throw new InvalidArgumentError('headers array must be even')
       }
       for (let i = 0; i < headers.length; i += 2) {
-        processHeader(this, headers[i], headers[i + 1])
+        const headerKey = headers[i]
+        const headerValue = headers[i + 1]
+        processHeader(this, headerKey, headerValue)
+        this[kRawRequestHeaders] = {
```

Comment:
can you explain this part to me? This seems to only add the last header pair whereas below all of the headers are stored

Reply:
Hmm, let me check something as I believe it will have problems with the headers as an array.

But in theory, what it should do is:

1. If the `headers` are passed as an array, it will iterate as usual, process them, and store them as an `object`. This is due to how `node:http2` expects the request headers.
2. If headers are passed as `object`, we allow the flow to continue as is, and stores the headers as plain object within the symbol.

Let me test the headers as an array, and harden the headers to make a copy of them instead of attaching the reference

Reply:
This will affect non h2 negatively.

Reply:
I'd prefer not to change the code here and instead do whaterver is need for h2 as close to h2 as possible

Reply:
yes, maybe my question wasn't worded good... right now the code is equal to:

```js
for (let i = 0; i < headers.length; i += 2) {
 // process headers
}

this[kRawRequestHeaders] = {
  [headers.at(-2)]: headers.at(-1)
}
```

while below, kRawRequestHeaders is set to the entire headers object. I don't understand the purpose of doing that, could you explain it to me?

Reply:
>I'd prefer not to change the code here and instead do whaterver is need for h2 as close to h2 as possible

The biggest issue comes with the processing of the header, as it gets stuck early in the request lifecycle, but maybe there's something I'm missing to make it closer to H2 implementation 👍 


>while below, kRawRequestHeaders is set to the entire headers object. I don't understand the purpose of doing that, could you explain it to me?

I'm seeing more clearly what are you suggestion, but let me know if I'm still beyond what you are suggesting

The change is not that far from the current implementation.
Users can provide the headers in a form of an array with a pair length.
When iterating over the pair length, we move forward two positions, and we take the `n` position as the header name, and `n+1` as the value.
The change I made here was merely to just store them as variables so they can be used to construct the headers object.

In the implementation below, the user passed the headers. We just iterate over its keys to validate each header and its value, 

If everything passes correctly, we just attach them directly into the `kRawRequestHeaders`.

Although I noticed an optimization here as the `processHeader` is already of the `this` context 😅

Nonetheless, I'll explore how to couple that part directly into H2 implementation